src/_pytest/unittest.py

```diff
@@ -182,6 +182,8 @@ def _handle_skip(self):
         return False
 
     def runtest(self):
+        if self.config.pluginmanager.get_plugin("pdbtrace"):
```

Comment:
something about this change sits very wrong with me i will require a deeper review of the interaction, cause it just looks "wrong"  - it cant be right to just invoke the hook that way

Reply:
after tracing the code for the trace plug-in and the debugging plug-in as well as the debugging plug-in, i believe this code will trigger double test execution when tracing

Reply:
Thanks for the insight @RonnyPfannschmidt . Could you elaborate on why or how this triggers double test execution, how I can reproduce it, and possibly some ideas on how I can go about tackling this issue?

Reply:
the double code execution happens due to **in addition** to running the normal test code below, when tracing calling the hook

i believe calling the hook itself is not fitting the intent of the unittest integration (as we explicitly invoke the testcase machinery of unittest for setup/teardown behaviour)

so in addition of the double test call, i believe the pyfunc_call based code additionally also skips the unittest setup/teardown machinery for the per test "scope"

to be honest without a own deeper investigation i have no idea where to actually hook in to make this "right"

Reply:
Got it. Let me play with this over the weekend and see if I can come up with something.

Reply:
@RonnyPfannschmidt I want to clarify the behavior of the trace plugin when the test case has a setUp() method. Let's say I have the following test case:
```
1 import unittest
2 class MyTest(unittest.TestCase):
3     def setUp(self):
4         assert True
5     def test_1(self):
6         assert True
```
When you run pytest with trace option, should this break on line 4, or line 6? 

What I find interesting is that if we have a pytest function with a fixture, the break starts at the function, not the fixture. For example, let's say we have the following test:
```
1 import pytest
2 @pytest.fixture
3 def setup():
4     assert True
5 def test_1(setup):
6     assert True
```
Running pytest with trace option will break at line 6. If we want to be consistent, running the trace option on a unittest style test case should break at the actual test, not the setup. Is this the correct behavior?

Reply:
yes, break at the start of the test is the intended behaviour